### PR TITLE
[IOTDB-2367] Linear fill should log warning for unsupported data type

### PR DESCRIPTION
**Pull request by CRZbulabula**

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSet.java [hunk 1]
```diff
@@ -174,13 +177,18 @@ public RowRecord nextWithoutConstraint() throws IOException {
...
     } catch (QueryProcessException e) {
-      logger.warn("group by fill has an exception while sliding: ", e);
+      logger.error("group by fill has an exception while sliding: ", e);
     }
 
     return record;
   }
 
   private void fillRecord(int index, RowRecord record) throws IOException {
+    if (unsupportedFillMethod[index]) {
```
Comment (ericpai) [inline server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSet.java:187]: It's better to add test cases to cover unsatisfied data types with `GROUP BY FILL`.

File: server/src/main/java/org/apache/iotdb/db/query/executor/FillQueryExecutor.java [hunk 2]
```diff
@@ -108,6 +113,15 @@ public QueryDataSet execute(QueryContext context)
 
         IFill fill = fillExecutors[i];
 
+        if (fill instanceof LinearFill
```
Comment (ericpai) [inline server/src/main/java/org/apache/iotdb/db/query/executor/FillQueryExecutor.java:116]: It's better to add test cases to cover unsatisfied data types with `FILL`.

**Review by ericpai — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSet.java (hunk 1 above)
Comment (CRZbulabula) [reply server/src/main/java/org/apache/iotdb/db/query/dataset/groupby/GroupByFillDataSet.java:187]: I've added tests in last commit

File: server/src/main/java/org/apache/iotdb/db/query/executor/FillQueryExecutor.java (hunk 2 above)
Comment (CRZbulabula) [reply server/src/main/java/org/apache/iotdb/db/query/executor/FillQueryExecutor.java:116]: I've added tests in last commit

**Review by ericpai — APPROVED** [review]

**Pull request merged**